`@Nullable` annotations on `equals` and collection methods.

cpovirk opened the pull request:
Aside from the annotations on `Annotation` (which was not previously
annotated), the other changes here were present in the old stubs but
not propagated to the new stubs:

https://github.com/typetools/checker-framework/blob/869caaab3f2b9c26600f394562dae30099fbc861/checker/jdk/nullness/src/java/util/EnumMap.java#L214
("Attempts to test for the presence of a null key or to remove one will,
however, function properly.")

https://github.com/typetools/checker-framework/blob/869caaab3f2b9c26600f394562dae30099fbc861/checker/jdk/nullness/src/java/util/Map.java#L465

Kakarot-SSJ4 commented:
@kelloggm @panacekcz @the1derer

the1derer commented:
@Kakarot-SSJ4 Can you please add comments related to Checker Framework using [`@CFComment`](https://checkerframework.org/api/org/checkerframework/framework/qual/CFComment.html)

the1derer commented on src/java.base/share/classes/java/util/zip/GZIPOutputStream.java:
```diff
@@ -182,10 +184,9 @@ public void finish() throws IOException {
-    */
+    @SuppressWarnings("cast.unsafe")
+    @CFComment("index: #1: cast from @IntVal(35615) int to @IntVal(31) byte will not produce any runtime error")
+    @CFComment("index: #2: cast from @IntVal(139) int to @IntVal(-117) byte will not produce any runtime error")
```
`@CFComment` is not repeatable you can write the comment as:
```
@CFComment({"index: #1: cast from @IntVal(35615) int to @IntVal(31) byte will not produce any runtime error","index: #2: cast from @IntVal(139) int to @IntVal(-117) byte will not produce any runtime error"})
```

Kakarot-SSJ4 replied:
Thank you for pointing it out, I've made that change now.

mernst commented:
Once https://github.com/typetools/checker-framework/pull/2714 is merged, please make any changes to this pull request as well.

the1derer commented:
@Kakarot-SSJ4 Please change your PR's base to `master` branch and then remove any Merge Conflicts.
Thanks.

mernst commented:
> @Kakarot-SSJ4 Please change your PR's base to `master` branch and then remove any Merge Conflicts.

Why does the base need to be changed?

the1derer commented:
> > @Kakarot-SSJ4 Please change your PR's base to `master` branch and then remove any Merge Conflicts.
> 
> Why does the base need to be changed?

Because `annotations-from-java-files` has already been merged to the master and it would be easier if all the new annotations now should be merged directly to master.

mernst commented:
> > > @Kakarot-SSJ4 Please change your PR's base to `master` branch and then remove any Merge Conflicts.
> > 
> > 
> > Why does the base need to be changed?
> 
> Because `annotations-from-java-files` has already been merged to the master and it would be easier if all the new annotations now should be merged directly to master.

You said that pull request was ready to be merged.  This indicates it was not complete and therefore was not ready to be merged.  Let's try to keep track of tasks.  A goal of the branches was to minimize the number of painful merges (and retargeting of pull requests).

Also, if the `annotations-from-java-files` branch serves no more purpose, it should be deleted.  I thought it was being retained for pull requests like this one.  Please clarify its status.

the1derer commented:
> You said that pull request was ready to be merged. This indicates it was not complete and therefore was not ready to be merged. Let's try to keep track of tasks. A goal of the branches was to minimize the number of painful merges (and retargeting of pull requests).

Yes, #58  was ready to be merged as PR corresponding to this PR i.e. typetools/checker-framework#2714 is still open and has not been merged. So, we were correct to merge #58. And this PR and all the future ones can be made directly to `master` branch.

> Also, if the `annotations-from-java-files` branch serves no more purpose, it should be deleted. I thought it was being retained for pull requests like this one. Please clarify its status.

Yes,  `annotation-from-java-files` branch can be deleted.

cpovirk committed:
`@Nullable` annotations on `equals` and collection methods.

Aside from the annotations on `Annotation` (which was not previously
annotated), the other changes here were present in the old stubs but
not propagated to the new stubs:

https://github.com/typetools/checker-framework/blob/869caaab3f2b9c26600f394562dae30099fbc861/checker/jdk/nullness/src/java/util/EnumMap.java#L214
("Attempts to test for the presence of a null key or to remove one will,
however, function properly.")

https://github.com/typetools/checker-framework/blob/869caaab3f2b9c26600f394562dae30099fbc861/checker/jdk/nullness/src/java/util/Map.java#L465

cpovirk commented:
Low-priority bump for this and https://github.com/typetools/jdk/pull/51

mernst commented:
Thanks for these annotations.
I'm sorry we didn't copy them properly from the stubs.
Also, my apologies for letting this pull request languish.  I wasn't watching this repository (maybe a slip of the mouse at some point?), so I received no notifications about your pull requests.
